src/vuestic-theme/vuestic-components/vuestic-modal/VuesticModalNew.vue

```diff
@@ -0,0 +1,263 @@
+<template>
+<div
+  v-show="valueProxy"
+  class="vuestic-modal-new__overlay"
+  @click="checkOutside"
+>
+  <transition name="modal">
+    <div
+      class="vuestic-modal-new"
+      :class="computedClass"
+      v-show="valueProxy"
+      :style="{maxWidth, maxHeight}"
+    >
+      <i
+        v-if="closeButton || fullscreen"
+        @click="cancel"
+        class="ion ion-md-close vuestic-modal-new__close"
+      />
+      <div class="vuestic-modal-new__inner" :style="{maxHeight, maxWidth}">
+        <div v-if="title" class="vuestic-modal-new__title">{{title}}</div>
+        <div v-if="hasHeaderSlot" class="vuestic-modal-new__header"><slot name="header"/></div>
+        <div v-if="message" class="vuestic-modal-new__message">{{message}}</div>
+        <div class="vuestic-modal-new__content">
+          <slot/>
+        </div>
+        <div v-if="cancelText !== '' || okText !== ''" class="vuestic-modal-new__actions">
+          <button v-if="cancelText !== ''" class="btn btn-secondary btn-micro" @click="cancel">{{cancelText}}</button>
+          <button class="btn btn-primary btn-micro" @click="ok">{{okText}}</button>
+        </div>
+      </div>
+    </div>
+  </transition>
+</div>
+</template>
+
+<script>
+export default {
+  name: 'vuestic-modal-new',
+  props: {
+    value: {
+      required: true,
+      default: false
+    },
+    position: {
+      type: String,
+      validator: function (value) {
+        return ['center', 'top', 'right', 'bottom', 'left'].indexOf(value) !== -1
+      }
+    },
+    title: String,
+    message: String,
+    okText: {
+      type: String,
+      default: 'OK'
+    },
+    cancelText: {
+      type: String,
+      default: 'Cancel'
+    },
+    closeButton: {
+      type: Boolean,
+      default: true
+    },
+    fullscreen: Boolean,
+    mobileFullscreen: {
+      type: Boolean,
+      default: true
+    },
+    noOutsideDismiss: Boolean,
+    noEscDismiss: Boolean,
+    maxWidth: {
+      type: String,
+      default: '600px'
+    },
+    maxHeight: {
+      type: String,
+      default: '100vh'
+    },
+    fixedLayout: Boolean,
+    onOk: Function,
+    onCancel: Function
+  },
+  computed: {
+    valueProxy: {
+      get () {
+        return this.value
+      },
+      set (value) {
+        this.$emit('input', value)
+      },
+    },
+    computedClass () {
+      return {
+        'vuestic-modal-new_fullscreen': this.fullscreen,
+        'vuestic-modal-new_mobile-fullscreen': this.mobileFullscreen,
+        [`vuestic-modal-new_position-${this.position}`]: this.position,
+        'vuestic-modal-new_fixed-layout': this.fixedLayout
+      }
+    },
+    hasHeaderSlot () {
+      return this.$slots.header
+    }
+  },
+  watch: {
+    valueProxy (value) {
+      if (value) {
+        window.addEventListener('keyup', this.listenKeyUp)
```

Comment:
What if we use in the template `@keyup.esc="onEsc` instead of registering / unregistering event listener?
@asvae , I think it should work if we put it on div, thoughts?

Reply:
Not going to work on div unless it is focused, sorry.